dexcom-analysis.py

```diff
@@ -0,0 +1,195 @@
+    df['count'] = 1
+
+    # The date does not matter here, since we just grab `.time`
+    dti = pd.date_range('1970-1-1', periods=periods, freq=bucket_size).time
+
+    # Excludes small amounts of carbs, which are likely snacks/corrections that are not paired with insulin
+    df = df.loc[df.carbs >= 5.0]
+
+    carbs = df.groupby(df['datetime'].dt.floor(bucket_size).dt.time)['count'].sum().reindex(dti, fill_value=0)
+    carbs = pd.DataFrame(carbs).reset_index()
+    carbs['datetime'] = pd.to_datetime(carbs['datetime'], format='%H:%M:%S')
+    
+    chart_carbs = alt.Chart(carbs, title=alt.Title("Frequency of Meal or Snack Time", fontSize=24)).mark_bar().encode(
+        x = alt.X('hoursminutes(datetime):O', axis=alt.Axis(labelAngle=-90), title="Time"),
+        y = alt.Y('count:Q', title="Count"),
+        tooltip=[alt.Tooltip('hoursminutes(datetime):O', title='Time'), alt.Tooltip('count', title='Count')]
+    ).properties(
+        width=screen_width-600,
+        height=screen_heigth-400,
+    )
+        
+    # carbs_plot = carbs.plot.bar(x="datetime", y="index", title="Frequency of Meal or Snack Time", color=['green'])
+    # plt.title("Frequency of Meal or Snack Time", fontsize = 24)
+    # carbs_plot.set_xlabel("Time", fontsize=16)
+    # carbs_plot.set_ylabel("Count", fontsize=16)
+    # # plt.show(block=True)
+    # fig = plt.gcf()
+    # fig.set_size_inches(18.5, 10.5)
+    # fig.savefig('out/meal_hist.png', dpi=100)
+    # plt.clf()
+    
+    return chart_carbs
+
+def main():
+    # outdir = './out'
+    # if not os.path.exists(outdir):
+    #     os.mkdir(outdir)
+
+    # Parse a CSV file into a base dataframe
+    try:
+        df = pd.read_csv('data/export.csv', index_col=False)
+    except FileNotFoundError as e:
+         df = pd.read_csv('data/sample.csv', index_col=False)
+
+    df.columns = [c.lower().replace(' ', '_') for c in df.columns]
+    df.drop(columns=['index', 'patient_info', 'device_info', 'source_device_id', 'glucose_rate_of_change_(mg/dl/min)', 'transmitter_time_(long_integer)', 'transmitter_id'], inplace=True, errors='ignore')
+    df.rename(columns = {'timestamp_(yyyy-mm-ddthh:mm:ss)':'datetime'}, inplace = True)
+    df['datetime'] = pd.to_datetime(df["datetime"])
+    df.sort_values(by='datetime', inplace = True)
+
+    # Create new dataframes with different types of events
+    insulin_events = df.loc[df.event_type == "Insulin"].copy()
+    carb_events = df.loc[df.event_type == "Carbs"].copy()
+    blood_glucose_events = df.loc[df.event_type == "EGV"].copy()
+
+    # Create plots
+    glucose = mean_glucose_plot(blood_glucose_events)
+    insuline = analyze_insulin(insulin_events, 48, '30T')
+    carbs = analyze_carbs(carb_events, 48, '30T')
+    
+    charts = alt.vconcat(glucose, insuline, carbs).to_html()
+    
+    # Display plots
+    webview.create_window('Average Blood Glucose', html=charts, width=screen_width, height=screen_heigth-50)
```

Comment:
```suggestion
    webview.create_window('Dexcom Data Analysis', html=charts, width=screen_width, height=screen_heigth-50)
```
Something to describe all the charts